cmd/maya-apiserver/cstor-operator/cspc/handler.go

```diff
@@ -137,6 +143,29 @@ func (c *Controller) syncCSPC(cspc *apis.CStorPoolCluster) error {
 		return nil
 	}
 
+	if !cspc.DeletionTimestamp.IsZero() {
+
+		// get all the BDCs associated with the cspc
+		bdcList, err := c.getUsedBDCs(cspc, openebsNameSpace)
+		if err != nil {
+			return err
+		}
+
+		// iterate over the bdc and remove the finalizer
+		for _, bdc := range bdcList.Items {
+			err := c.removeBDCFinalizer(&bdc, v1alpha1.CSPCFinalizer)
+			if err != nil {
+				return err
+			}
+		}
+
+		// remove finalizer on cspc
+		err = c.removeCSPCFinalizer(cspc, v1alpha1.CSPCFinalizer)
+		if err != nil {
+			return err
+		}
```

Comment:
we need to be returning from here to avoid reconcilation

Reply:
Done